lib/ansible/modules/monitoring/zabbix/zabbix_host.py

```diff
@@ -413,6 +465,10 @@ def update_host(self, host_name, group_ids, status, host_id, interfaces, exist_i
                 parameters['ipmi_username'] = ipmi_username
             if ipmi_password:
                 parameters['ipmi_password'] = ipmi_password
+            if macros:
```

Comment:
Also: how are we supposed to remove/clear macros and tags?
I tried

```yaml
tags: []
macros: []
```

with and without `force: True` but it didn't remove anything.

As I understand it, it's because your are using just a `if macros:` in update_host function:

```python
            if macros:
                parameters['macros'] = macros
            if tags:
                parameters['tags'] = tags
```

this way even if we set `macros: []` in config it doesn't get in the actual parameters. In my opinion an `if macros is not None:` would be better. Similar for tags.

Reply:
Good catch. True

Reply:
Now noted in module doc that providing empty list with `force=yes` will wipe all of existing tags/macros.